src/sentry_plugins/github/plugin.py

```diff
@@ -399,3 +409,141 @@ def compare_commits(self, repo, start_sha, end_sha, actor=None):
                 self.raise_error(e)
             else:
                 return self._format_commits(repo, res['commits'])
+
+
+class GitHubAppsRepositoryProvider(GitHubRepositoryProvider):
+    name = 'GitHub Apps'
+    auth_provider = 'github_apps'
+    logger = logging.getLogger('sentry.plugins.github_apps')
+
+    def get_install_url(self):
+        return options.get('plugins.github.apps-install-url')
+
+    def get_available_auths(self, user, organization, integrations, social_auths, **kwargs):
+        allowed_gh_installations = set(self.get_installations(user))
+
+        linked_integrations = {i.id for i in integrations}
+
+        _integrations = list(
+            Integration.objects.filter(
+                external_id__in=allowed_gh_installations,
+            )
+        )
+
+        # add in integrations that might have been set up for org
+        # by users w diff permissions
+        _integrations.extend(
+            [i for i in integrations if i.external_id not in allowed_gh_installations]
+        )
+
+        return [
+            {
+                'defaultAuthId': None,
+                'user': None,
+                'externalId': i.external_id,
+                'integrationId': six.text_type(i.id),
+                'linked': i.id in linked_integrations,
+            } for i in _integrations
+        ]
+
+    def link_auth(self, user, organization, data):
+        integration_id = data['integration_id']
+
+        try:
+            integration = Integration.objects.get(
+                provider=self.auth_provider,
+                id=integration_id,
+            )
+        except Integration.DoesNotExist:
+            raise PluginError('Invalid integration id')
+
+        # check that user actually has access to add
+        allowed_gh_installations = set(self.get_installations(user))
+        if int(integration.external_id) not in allowed_gh_installations:
+            raise PluginError('You do not have access to that integration')
+
+        integration.add_organization(organization.id)
+
+        for repo in self.get_repositories(integration):
+            # TODO(jess): figure out way to migrate from github --> github apps
```

Comment:
it wouldn't be difficult to do this, just wondering if we should, and if so, if i should also delete the individual repo's webhook in favor of using the integration's